github_pr_to_internal_pr/github_pr_to_internal_pr.py

```diff
@@ -62,10 +52,21 @@ def setup_project(project_fullname):
     gl.auth()
 
     HDR_LEN = 8
-    gl_project_url = GITLAB_URL[: HDR_LEN] + GITLAB_TOKEN + ':' + GITLAB_TOKEN + '@' + GITLAB_URL[HDR_LEN :] + '/' + project_fullname + '.git'
+    gl_project_url = GITLAB_URL[:HDR_LEN] + GITLAB_TOKEN + ':' + GITLAB_TOKEN + '@' + GITLAB_URL[HDR_LEN:] + '/' + repo_fullname + '.git'
 
     print('Cloning repository...')
-    Git(".").clone(gl_project_url, recursive=True)
+    Git('.').clone(project_html_url, branch='master', single_branch=True, recursive=True)
+
+    git = Git(project_name)
+    GITLAB_REMOTE = 'gitlab'
+
+    print('Adding and fetching the internal remote...')
+    git.remote('add', GITLAB_REMOTE, gl_project_url)
+    git.pull(GITLAB_REMOTE, 'master')
+
+    print('Updating submodules...')
+    git.submodule('update', '--init', '--recursive')
```

Comment:
Question, is it really necessary to initialize submodule contents? I think merging or rebasing will work even without this step.

Reply:
@igrr What if PR has modification to submodule itself, like updating third library pointer?

Reply:
I have tested the action locally (submodule as changes in PR) and it works as expected. Resolving the comment.